spring-integration-core/src/main/java/org/springframework/integration/config/AggregatorFactoryBean.java

```diff
@@ -48,128 +49,267 @@
  *
  */
 public class AggregatorFactoryBean extends AbstractSimpleMessageHandlerFactoryBean<AggregatingMessageHandler>
-				implements ApplicationContextAware, BeanNameAware, ApplicationEventPublisherAware {
+		implements ApplicationContextAware, BeanNameAware, ApplicationEventPublisherAware {
 
-	private final AggregatingMessageHandler aggregator;
+	private Object processorBean;
 
-	public AggregatorFactoryBean(Object processor) {
-		this(processor, null);
+	private String methodName;
+
+	private ApplicationContext applicationContext;
+
+	private String beanName;
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	private AggregatingMessageHandler aggregator;
```

Comment:
OK. `AbstractSimpleMessageHandlerFactoryBean` has `handler` property on the matter.
Doesn't that work for you ?